corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -328,6 +375,47 @@ def save_case(case):
                     logging.debug(msg)
                 raise CaseSaveError(e)
 
+    @staticmethod
+    def get_open_case_ids(domain, owner_id):
+        owner_id_is_falsey = Q(owner_id=None) | Q(owner_id="")
+        owner_query = Q(owner_id=owner_id) | (owner_id_is_falsey & Q(modified_by=owner_id))
+        query = CommCareCaseSQL.objects.filter(domain=domain, closed=False).filter(owner_query)
+        return list(query.values_list('case_id', flat=True))
+
+    @staticmethod
+    def get_closed_case_ids(domain, owner_id):
+        query = CommCareCaseSQL.objects.filter(domain=domain, owner_id=owner_id, closed=True)
+        return list(query.values_list('case_id', flat=True))
+
+    @staticmethod
+    def get_case_ids_modified_with_owner_since(domain, owner_id, reference_date):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_extension_case_ids(domain, case_ids): # I need like this but without the relationship id check
+        """
+        Given a base list of case ids, for those that are open, get all ids of all extension cases that reference them
+        """
+        query = CommCareCaseIndexSQL.objects.filter(
+            # domain=domain,  # TODO: Why isn't domain being used?
+            referenced_id__in=case_ids,
+            relationship_id=CommCareCaseIndexSQL.EXTENSION
+        )
+        # TODO: Filter by open cases?
```

Comment:
Seems you need to join on the the case table anyway so you can add the domain check there